File: java-app/src/main/java/com/prosoft/java/service/UserService.java
package com.prosoft.java.service;

import com.prosoft.java.domain.User;
import com.prosoft.java.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Collections;
import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    private final UserRepository userRepository;

    @Autowired
    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public List<User> getUser(Long id) {
        return Optional.ofNullable(id)
                .flatMap(userRepository::findById)
                .map(Collections::singletonList)
                .orElse(Collections.emptyList());
    }

}

File: java-app/src/main/kotlin/com/prosoft/kotlin/service/UserService.kt
package com.prosoft.kotlin.service

import com.prosoft.kotlin.domain.User
import com.prosoft.kotlin.repository.UserRepository
import org.springframework.stereotype.Service

@Service("kotlinUserService")
class UserService(private val userRepository: UserRepository) {

    /**
     * "id?.let {...}" - исп. безопасный вызов ?., чтобы проверить, что id не null, и только в этом случае выполнить поиск.
     * "userRepository.findById(it)" - ищем пользователя по id, результат — либо найденный пользователь, либо null.
     * "?.let { listOf(it) }" - если пользователь найден, оборачиваем его в список с помощью listOf.
     * "?: emptyList()" - если пользователя нет, возвращаем пустой список.
     */
    fun getUser(id: Long?): List<User> {
        return id?.let { userRepository.findById(it) }
            ?.let { listOf(it) }
            ?: emptyList()
    }

}